common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -209,9 +209,7 @@ bool socket_thread::verify_cmdu(message::sUdsHeader *uds_header)
             if (tlv_vendor_specific.vendor_oui() ==
                 ieee1905_1::tlvVendorSpecific::eVendorOUI::OUI_INTEL) {
                 // assuming that the magic is the first data on the beerocks header
-                auto beerocks_magic = *(
-                    uint32_t *)((uint8_t *)tlv + ieee1905_1::tlvVendorSpecific::get_initial_size());
-
+                auto beerocks_magic = *(uint32_t *)(tlv_vendor_specific.payload());
```

Comment:
We should get rid of the magic in the beerocks action header. It is not relevant anymore, and it is basically a leftover of the area before we used cmdu.

Reply:
It still works and it still validates the TLV, so I don't think removing it now is beneficial.
Removing this will be done as part of moving verify_cmdu to the transport.